AIMSICD/src/main/java/com/secupwn/aimsicd/ui/fragments/MapFragment.java

```diff
@@ -289,28 +291,37 @@ private void setupMapType(int mapType) {
      */
     private void setUpMapIfNeeded() {
 
+        mMap.getOverlayManager().clear();
         // Check if we were successful in obtaining the map.
         mMap.setBuiltInZoomControls(true);
         mMap.setMultiTouchControls(true);
         mMap.setMinZoomLevel(3);
         mMap.setMaxZoomLevel(19); // Latest OSM can go to 21!
         mMap.getTileProvider().createTileCache();
-        mCompassOverlay = new CompassOverlay(getActivity(), new InternalCompassOrientationProvider(getActivity()), mMap);
+
+        if (null == mCompassOverlay) {
```

Comment:
Could you keep the `null` and other values on the right of logical operators to keep the code consistent?